poc_mobile_verifier/src/subnetwork_rewards.rs

```diff
@@ -92,600 +49,30 @@ impl SubnetworkRewards {
-                    .clone()
-                    .into_iter()
-                    .map(|(key, weight)| proto::OwnerShare {
-                        pub_key: key.to_vec(),
-                        weight: bones_to_u64(weight),
-                    })
-                    .collect(),
-            },
-        )?;
-
-        file_sink::write(
-            shares_tx,
-            proto::Shares {
-                shares: shares
-                    .into_iter()
-                    .map(|(cbsd_id, share)| proto::Share {
-                        cbsd_id,
-                        timestamp: share.timestamp.timestamp() as u64,
-                        pub_key: share.pub_key.to_vec(),
-                        weight: bones_to_u64(share.weight),
-                        cell_type: share.cell_type as i32,
-                        validity: proto::ShareValidity::Valid as i32,
-                    })
-                    .collect(),
-            },
-        )
-        .await?;
-
-        file_sink::write(
-            invalid_shares_tx,
-            proto::Shares {
-                shares: invalid_shares,
-            },
-        )
-        .await?;
-
+        subnet_rewards_tx: &file_sink::MessageSender,
+    ) -> file_store::Result<oneshot::Receiver<file_store::Result>> {
         file_sink::write(
-            subnet_tx,
+            subnet_rewards_tx,
             proto::SubnetworkRewards {
-                start_epoch: after_ts,
-                end_epoch: before_ts,
-                rewards,
+                start_epoch: self.epoch.start.timestamp() as u64,
+                end_epoch: self.epoch.end.timestamp() as u64,
+                rewards: self.rewards,
             },
         )
-        .await?;
-
-        Ok(())
-    }
-}
-
-fn write_json(
-    fname_prefix: &str,
-    after_ts: u64,
-    before_ts: u64,
-    data: &impl serde::Serialize,
-) -> Result {
-    let tmp_output_dir = std::env::var("TMP_OUTPUT_DIR").unwrap_or_else(|_| "/tmp".to_string());
-    let fname = format!("{}-{}-{}.json", fname_prefix, after_ts, before_ts);
-    let fpath = Path::new(&tmp_output_dir).join(&fname);
-    std::fs::write(Path::new(&fpath), serde_json::to_string_pretty(data)?)?;
-    Ok(())
-}
-
-#[cfg(test)]
-mod test {
-    use super::proto::ShareValidity;
-    use crate::{
-        cell_type::CellType,
-        reward_share::{OwnerEmissions, OwnerResolver, Share, Shares},
-        reward_speed_share::{SpeedShare, SpeedShareMovingAvgs, SpeedShares},
-    };
-    use async_trait::async_trait;
-    use chrono::TimeZone;
-    use helium_crypto::PublicKey;
-    use std::str::FromStr;
-
-    use super::*;
-
-    struct FixedOwnerResolver {
-        owner: PublicKey,
-    }
-
-    #[async_trait]
-    impl OwnerResolver for FixedOwnerResolver {
-        async fn resolve_owner(&mut self, _address: &PublicKey) -> Result<Option<PublicKey>> {
-            Ok(Some(self.owner.clone()))
-        }
-    }
-
-    #[tokio::test]
-    async fn check_rewards() {
```

Comment:
can these tests not be salvaged and preserved, even with some conversion?

Reply:
they are preserved, they got moved to `reward_share.rs` (which I think is a more appropriate place for them)